crates/sncast/src/starknet_commands/declare.rs

```diff
@@ -57,11 +59,35 @@ impl_payable_transaction!(Declare, token_not_supported_for_declaration,
     DeclareVersion::V3 => FeeToken::Strk
 );
 
-#[allow(clippy::too_many_lines)]
-pub async fn declare(
+impl From<&DeclareDeploy> for Declare {
+    fn from(declare_deploy: &DeclareDeploy) -> Self {
+        let DeclareDeploy {
+            contract_name,
+            deploy_args: DeployArgs { package, .. },
+            fee_token,
+            rpc,
+        } = &declare_deploy;
+
+        let fee_args = FeeArgs {
+            fee_token: Some(fee_token.to_owned()),
+            ..Default::default()
+        };
+
+        Declare {
+            contract: contract_name.to_owned(),
+            fee_args,
+            nonce: None,
+            package: package.to_owned(),
+            version: None,
```

Comment:
Is it okay to set these to `None`. I'm assuming they are being set to correct values somewhere down the line.

Reply:
Yeah, `version` is going to be inferred from `FeeToken` specified in `fee_args`